lib/sample_uploader/utils/importer.py

```diff
@@ -212,77 +237,133 @@ def import_samples_from_file(
+            df = df.rename(columns=column_mapping)
+        # redundant, even harmful if things get out of sync
+        # verify_columns(df)
+        for key in column_mapping:
+            if key in columns_to_input_names:
+                val = columns_to_input_names.pop(key)
+                columns_to_input_names[column_mapping[key]] = val
 
-    if params['file_format'].upper() in ['SESAR', "ENIGMA"]:
-        if 'material' in df.columns:
-            df.rename(columns={"material": params['file_format'].upper() + ":material"}, inplace=True)
-            val = columns_to_input_names.pop("material")
-            columns_to_input_names[params['file_format'].upper() + ":material"] = val
-    if params['file_format'].upper() == "KBASE":
-        if 'material' in df.columns:
-            df.rename(columns={"material": "SESAR:material"}, inplace=True)
-            val = columns_to_input_names.pop("material")
-            columns_to_input_names["SESAR:material"] = val
+        if params['file_format'].upper() in ['SESAR', "ENIGMA"]:
+            if 'material' in df.columns:
+                df.rename(columns={"material": params['file_format'].upper() + ":material"}, inplace=True)
+                val = columns_to_input_names.pop("material")
+                columns_to_input_names[params['file_format'].upper() + ":material"] = val
+        if params['file_format'].upper() == "KBASE":
+            if 'material' in df.columns:
+                df.rename(columns={"material": "SESAR:material"}, inplace=True)
+                val = columns_to_input_names.pop("material")
+                columns_to_input_names["SESAR:material"] = val
 
-    acls = {
-        "read": [],
-        "write": [],
-        "admin": [],
-        "public_read": -1  # set to false (<0)
-    }
-    if params.get('share_within_workspace'):
-        # query workspace for user permissions.
-        acls = get_workspace_user_perms(workspace_url, params.get('workspace_id'), token, username, acls)
-    groups = SAMP_SERV_CONFIG['validators']
+        acls = {
+            "read": [],
+            "write": [],
+            "admin": [],
+            "public_read": -1  # set to false (<0)
+        }
+        if params.get('share_within_workspace'):
+            # query workspace for user permissions.
+            acls = get_workspace_user_perms(workspace_url, params.get('workspace_id'), token, username, acls)
+        groups = SAMP_SERV_CONFIG['validators']
+
+        cols = list(set(df.columns) - set(REGULATED_COLS))
+        sample_url = get_sample_service_url(sw_url)
+        samples, existing_samples, produce_errors = _produce_samples(
+            df,
+            cols,
+            column_groups,
+            column_unit_regex,
+            sample_url,
+            token,
+            input_sample_set['samples'],
+            columns_to_input_names,
+            header_row_index
+        )
+        errors += produce_errors
+    
+    if params.get('prevalidate') and not errors:
```

Comment:
it is checking again that no errors were produce in the previous step. I could indent it, but it wouldn't change the if condition (because errors needs to be re-checked).

Reply:
Unless there is an alternative I'm not spotting...

Reply:
right, i guess I'm just confused on if there is a situation in which you do this step if you don't do the previous step

Reply:
There is not, no

Reply:
not a big deal for now